ddtrace/profiling/exporter/http.py

```diff
@@ -80,12 +84,17 @@ def _get_service_name():
             return service_name
 
 
+def _validate_api_key(instance, attribute, value):
+    if not formats.validate_api_key(value):
+        raise InvalidAPIKey("Invalid API key")
```

Comment:
I'm not 100% about hard-failing here. In the super unlikely event that the API key format changes the library will not be usable. How would we feel about a `warning` level log message?

Reply:
we do sometimes use invalid api keys for testing

Reply:
If you change the format, it'll work with a warning saying the API key is invalid. Not sure that makes your software "better quality" 😄

Reply:
This is true. Maybe we could be a bit more lax with the validation? I think the only check I have issue with is the length. I think I'd be more comfortable with `>= 16`, but just a thought. I think I'm okay with it either way.

Reply:
I stole https://github.com/DataDog/logs-backend/blob/prod/service/src/main/java/com/fsmatic/key/ApiKeys.java#L124

TBH I'm not even sure it's a good idea to do validation in the client. The problem is that the API just returns `400` without detail when the API key (or any field is invalid), so we have to come up with a client-side solution.

Reply:
Yeah the ideal solution would be to use a back-end response. Another approach is we could report in the exception/log that a common reason for a 400 is an invalid API key.

Reply:
I wouldn't validate in the client, if we ever have to change the format for an API key, then we'll need to keep this up to date.

The api call should fail, or blackhole, or however the API chooses to handle.

Reply:
I'm not sure I understand your suggestion @brettlangdon.
I mean the API does fail and return 400 with no detail so… are you suggesting to just drop this PR altogether?

@Kyle-Verhoog actually my first idea was to do exactly that poor version of saying "hey I got 400, maybe your API key is invalid".

Reply:
I mostly just mean to drop the validation part, but having some sort of warning/error, log message, back-off, retry... whatever is very reasonable.

Reply:
We already log the 400 and we got #1452 
Retrying isn't going to help in that case.
… so we're down with the option of logging "hey I got 400, maybe your API key is invalid".